core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/naming/generation/DefaultNameGenerator.java

```diff
@@ -0,0 +1,29 @@
+package uk.ac.bbsrc.tgac.miso.core.service.naming.generation;
+
+import org.apache.log4j.Logger;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Nameable;
+import uk.ac.bbsrc.tgac.miso.core.exception.MisoNamingException;
+import uk.ac.bbsrc.tgac.miso.core.service.naming.DefaultMisoEntityPrefix;
+import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
+
+public class DefaultNameGenerator implements NameGenerator<Nameable> {
+
+  private static final Logger log = Logger.getLogger(DefaultNameGenerator.class);
+
+  @Override
+  public String generate(Nameable nameable) throws MisoNamingException {
+    if (DefaultMisoEntityPrefix.get(nameable.getClass().getSimpleName()) == null) {
```

Comment:
Please file a ticket to replace this with a different interface and reference it here. This should be something like:
```
for (DefaultMisoEntity prefix : DefaultMisoEntity.values()) {
  if (prefix.getEntityClass().isAssignableFrom(nameable.getClass())) { return prefix.getEntityClass().name() + nameable.getId(); }
}
throw...
}
```